core/views.py

```diff
@@ -25,6 +28,38 @@ def get_object(self: "Post", queryset: QuerySet[Post] | None = None) -> Post:
         obj.update_display_counter()
         return obj
 
+    def get_context_data(self: "PostDetailView", **kwargs: dict[str, Any]) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+        comments = self.object.get_comments()
+        form = CommentForm()
+
+        context["comments"] = comments
+        context["form"] = form
+        return context
+
+    def post(self: "PostDetailView", request: HttpRequest, pk: int) -> HttpResponse:
+        post = get_object_or_404(Post, id=pk)
+        form = CommentForm(request.POST)
+        if form.is_valid():
+            new_comment = form.save(commit=False)
+            parent_id = request.POST.get("parent_id")
+            if parent_id:
+                new_comment.parent = Post.objects.get(id=parent_id)
+            else:
+                new_comment.parent = post
+            new_comment.author = request.user
+            new_comment.community = post.community
+            new_comment.save()
+            return redirect("post-detail", pk=pk)
+
+        comments = post.get_comments()
+        context = {
+            "post": post,
+            "comments": comments,
+            "form": form,
+        }
+        return render(request, "post_detail.html", context)
```

Comment:
Hmm I know we can use render but...
Cold you use
```
html_content = render_to_string("post_detail.html", context)
return HttpResponse(html_content)
```
instead?

Reply:
Also, on success it should redirect to view page, so on refresh there will be no "Are you sure you want to resubmit the form" question?